src/dataflow/src/source/kinesis.rs

```diff
@@ -217,7 +217,7 @@ impl SourceReader for KinesisSourceReader {
                         }
                     };
 
-                    for record in get_records_output.records.unwrap_or_else(|| vec![]) {
+                    for record in get_records_output.records.unwrap_or_else(Default::default) {
```

Comment:
ah good point!

Reply:
Clippy occasionally gets angry at `unwrap_or(...)` because it often doesn't know that `...` is zero cost to construct. E.g. if `...` were allocating or something, `unwrap_or_else` would be "better".

Reply:
I think in this case, it doesn't complain! I am would not surprised if clippy now knows that vec![] does not allocate

Reply:
but it DOES think `unwrap_or_default` is better, which is true!